tests/Doctrine/Tests/OrmFunctionalTestCase.php

```diff
@@ -946,4 +950,60 @@ final protected function getLastLoggedQuery(int $index = 0): array
 
         return $lastQuery;
     }
+
+    public function getLimitSQLByPlatform($limit, AbstractPlatform $platform): string
+    {
+        $sql = ' LIMIT';
+        if ($platform instanceof OraclePlatform) {
+            $sql = ' WHERE ROWNUM <=';
+        }
+
+        return $sql . ' ' . $limit;
+    }
+
+    public function getDiscriminatorByPlatform($key, AbstractPlatform $platform): string
```

Comment:
This looks like a hack. Aren't there non-test APIs for this? It shouldn't be public anyways.

Reply:
Unfortunately, I did not write these tests. I can't say why they decided to add them. I just brought it to mind for comparison, to. Oracle all converts to uppercase unless explicitly specified.

Reply:
The test is an example of a program that solves a problem using the API being tested. Does it mean that every user of the ORM now needs to implement such a function?